### PR TITLE
[dvsim tool] "use_cfgs" usecase enhancement

### PR DESCRIPTION
**Pull request by sriyerg**
- expanded "use_cfgs" to list hjson file targets or complete hjson dicts
inline
- this will allow specification of simpler, repeatitive cfgs such as
lint a lot more concisely

Signed-off-by: Srikrishna Iyer <sriyer@google.com>

### CODE REVIEW

File: util/dvsim/FlowCfg.py [hunk 1]
```diff
@@ -376,7 +453,7 @@ def publish_results_summary(self):
             md_results_to_html(self.results_title, css_path,
                                self.results_summary_md))
         f.close()
-        rm_cmd = "rm -rf " + results_html_file + "; "
+        rm_cmd = "/bin/rm -rf " + results_html_file + "; "
```
Comment (msfschaffner) [inline util/dvsim/FlowCfg.py]: why this change?

File: util/dvsim/FlowCfg.py [hunk 2]
```diff
@@ -210,12 +230,69 @@ def resolve_hjson_raw(self, hjson_dict):
...
+        name = idict["name"] if "name" in idict.keys() else None
+        if not name:
+            log.error(
+                "In-line entry in use_cfgs list does not contain \
+                      a \"name\" key (will be skipped!):\n%s", idict)
+            return None
+
+        # Check if temp cfg file already exists
+        temp_cfg_file = self.scratch_root + "/." + self.branch + "__" + \
+                        name + "_cfg.hjson"
```
Comment (msfschaffner) [inline util/dvsim/FlowCfg.py]: not sure, but doesn't this lead to result path conflicts in the publish functions later on? 

say, if we were to use this inline mechanism for multiple different flows like synthesis and lint, wouldn't the summaries overwrite each other since they live in the same base path?

Comment (sriyerg) [reply util/dvsim/FlowCfg.py]: The publish functions use `rel_path`, which is constructed in `__init__` from the path to this file. Since this is a 'fake' file that is generated to support this flow, we will need to move `rel_path` to the hjson instead. Let me think more on this.

File: util/dvsim/FlowCfg.py (hunk 1 above)
Comment (sriyerg) [reply util/dvsim/FlowCfg.py]: Some people have aliases for 'rm'.

File: util/dvsim/FlowCfg.py (hunk 2 above)
Comment (sriyerg) [reply util/dvsim/FlowCfg.py]: I made some more changes. Inline cfgs can now specify the `rel_path` to indicate where to publish the final results.

File: util/dvsim/FlowCfg.py [hunk 3]
```diff
@@ -364,47 +440,14 @@ def gen_results_summary(self):
         '''
         return
 
-    def publish_results_summary(self):
```
Comment (sriyerg) [inline util/dvsim/FlowCfg.py]: Moved this below.
